php-checks/src/main/java/org/sonar/php/checks/phpunit/NoAssertionInTestCheck.java

```diff
@@ -0,0 +1,163 @@
+      context().newIssue(this, tree.name(), MESSAGE);
+    }
+  }
+
+  private class AssertionsFindVisitor extends PHPVisitorCheck {
+    private boolean didFindAssertion = false;
+    private final SymbolTable symbolTable;
+
+    private AssertionsFindVisitor(SymbolTable symbolTable) {
+      this.symbolTable = symbolTable;
+    }
+
+    @Override
+    public void visitFunctionCall(FunctionCallTree tree) {
+      String functionName = getFunctionName(tree);
+
+      if (isAssertion(tree) ||
+        functionNameCountsAsAssertion(functionName) ||
+        isDynamicFunctionCall(tree) ||
+        isLocalMethodWithAssertion(tree)) {
+        didFindAssertion = true;
+      }
+
+      super.visitFunctionCall(tree);
+    }
+
+    private boolean functionNameCountsAsAssertion(@Nullable String functionName) {
+      if (functionName == null) {
+        return false;
+      }
+
+      return ASSERTION_METHODS_PATTERN.matcher(functionName).matches() ||
+        TEST_CONTROL_FUNCTIONS.contains(functionName.toLowerCase(Locale.ROOT));
+    }
+
+    private boolean isDynamicFunctionCall(FunctionCallTree tree) {
+      Tree functionNameTree = tree.callee();
+
+      if (functionNameTree.is(Tree.Kind.CLASS_MEMBER_ACCESS, Tree.Kind.OBJECT_MEMBER_ACCESS)) {
+        functionNameTree = ((MemberAccessTree) functionNameTree).member();
+      }
+
+      return !functionNameTree.is(Tree.Kind.NAMESPACE_NAME, Tree.Kind.NAME_IDENTIFIER);
+    }
+
+    private boolean isLocalMethodWithAssertion(FunctionCallTree tree) {
+      MethodDeclarationTree methodDeclaration;
+      Optional<MethodDeclarationTree> optionalMethodDeclaration = getMethodDeclarationTree(tree);
+      if (optionalMethodDeclaration.isPresent()) {
+        methodDeclaration = optionalMethodDeclaration.get();
+      } else {
+        return false;
+      }
+
+      if (!assertionInMethod.containsKey(methodDeclaration)) {
+        assertionInMethod.put(methodDeclaration, false);
+        AssertionsFindVisitor v = new AssertionsFindVisitor(symbolTable);
+        methodDeclaration.accept(v);
+        assertionInMethod.put(methodDeclaration, v.didFindAssertion);
+      }
+      return assertionInMethod.get(methodDeclaration);
+    }
+
+    private Optional<MethodDeclarationTree> getMethodDeclarationTree(FunctionCallTree tree) {
+      ExpressionTree callee = tree.callee();
+      if (!callee.is(Tree.Kind.CLASS_MEMBER_ACCESS, Tree.Kind.OBJECT_MEMBER_ACCESS)) {
+        return Optional.empty();
+      }
+
+      Symbol symbol = symbolTable.getSymbol(((MemberAccessTree) callee).member());
+      if (symbol != null && symbol.is(Symbol.Kind.FUNCTION)) {
+        return Optional.ofNullable((MethodDeclarationTree) TreeUtils.findAncestorWithKind(symbol.declaration(),
+          ImmutableSet.of(Tree.Kind.METHOD_DECLARATION)));
+      }
+
+      return Optional.empty();
+    }
+
+    private @Nullable String getFunctionName(FunctionCallTree tree) {
+      String functionName = CheckUtils.getFunctionName(tree);
```

Comment:
I would recommend "getLowerCaseFunctionName" so you don't have to carry about the case-insensitivity.